plugins/modules/filesystem.py

```diff
@@ -215,26 +335,26 @@ def chfs(module, filesystem):
 
     opts = ""
 
+    # check initial attributes
+    changed = check_attr_change(module, filesystem)
+    if not changed:
+        msg = "No changes needed in %s" % filesystem
+        return False, msg
+
     if is_nfs(module, filesystem):
         # Modify NFS filesystem
         if amount is True:
             opts += "-A "
         elif amount is False:
             opts += "-a "
 
-        if device:
-            opts += "-d %s " % device
-
         if perms:
             opts += "-t %s " % perms
 
-        if nfs_server:
-            opts += "-h %s " % nfs_server
-
         if mgroup:
             opts += "-m %s " % mgroup
 
-        cmd = "chnfsmnt %s -f %s" % (opts, filesystem)
+        cmd = "chnfsmnt %s -f %s -d %s -h %s" % (opts, filesystem, device, nfs_server)
```

Comment:
device check is removed, and it is added directly to cmd line, what if no device exists, will this cause any issues?

Reply:
for chnfsmnt, device and nfs_server parameter is required. so even if i did not remove those checks if those parameters are not inputted then it will still give an error.

Reply:
I see. what is benefit by removing device?

Reply:
there was no point in checking those parameters when it is required (the checks did not error out, and relied on chnfsmnt to error out to report back to ansible so i just removed those checks and added them directly to the cmd to make it clear that those are required parameters), and this code path is only hit for when the FS in NFS type.